faunadb/stream_events.go

```diff
@@ -0,0 +1,165 @@
+package faunadb
+
+import (
+	"fmt"
+)
+
+// StreamEventType is a stream eveny type
+type StreamEventType = string
+
+const (
+	// ErrorEventT is the stream error event type
+	ErrorEventT StreamEventType = "error"
+
+	// HistoryRewriteEventT is the stream history rewrite event type
+	HistoryRewriteEventT StreamEventType = "history_rewrite"
+
+	// StartEventT is the stream start event type
+	StartEventT StreamEventType = "start"
+
+	// VersionEventT is the stream version event type
+	VersionEventT StreamEventType = "version"
+)
+
+// StreamEvent represents a stream event with a `type` and `txn`
+type StreamEvent interface {
+	Type() StreamEventType
+	Txn() int64
+	String() string
+}
+
+// StartEvent emitted when a valid stream subscription begins.
+// Upcoming events are guaranteed to have transaction timestamps equal to or greater than
+// the stream's start timestamp.
+type StartEvent struct {
+	StreamEvent
+	txn   int64
+	event int64
+}
+
+func unMarshalStreamEvent(data Obj) (evt StreamEvent, err error) {
```

Comment:
minor: I'd move this function down to after its constructed types have been declared.

Reply:
Done